src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php

```diff
@@ -87,16 +104,32 @@ class EmployeeClaimRequestAPI extends Endpoint implements CrudEndpoint
     public function create(): EndpointResult
     {
         $claimRequest = new ClaimRequest();
-        $this->setClaimRequest($claimRequest);
+        $empNumber = $this->getEmpNumber();
+
+        if (!$this->checkLoggedInUser($empNumber)) {
+            throw $this->getForbiddenException();
+        }
+
+        if (!$this->getUserRoleManagerHelper()->isEmployeeAccessible($empNumber)) {
+            throw $this->getForbiddenException();
+        }
```

Comment:
```suggestion
```

Reply:
is supervisor accessible to a ESS - this type of scenarios cannot handle without this